src/libopensc/sm.c

```diff
@@ -158,7 +158,16 @@ sc_sm_single_transmit(struct sc_card *card, struct sc_apdu *apdu)
 
 	/* send APDU flagged as NO_SM */
 	sm_apdu->flags |= SC_APDU_FLAGS_NO_SM;
-	rv = sc_transmit_apdu(card, sm_apdu);
+	/* if token is epass2003, using below code*/
+	switch(card->type){
+		case SC_CARD_TYPE_ENTERSAFE_FTCOS_EPASS2003:
```

Comment:
I'd like to avoid card specific code on this layer. What's the reason for this change?